src/core/featurechecklistproxymodel.h

```diff
@@ -0,0 +1,84 @@
+/***************************************************************************
+  featurechecklistproxymodel.h - FeatureCheckListProxyModel
+
+ ---------------------
+ begin                : Jun 2025
+ copyright            : (C) 2025 by Mohsen Dehghanzadeh
+ email                : mohsen (at) opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef FEATURECHECKLISTPROXYMODEL_H
+#define FEATURECHECKLISTPROXYMODEL_H
+
+#include <QSortFilterProxyModel>
+
+class FeatureCheckListProxyModel : public QSortFilterProxyModel
```

Comment:
Can we add a bit of documentation for the class itself? See e.g.:

https://github.com/opengisch/QField/blob/master/src/core/messagelogmodel.h#L23-L28